### PR TITLE
Revert "revert all 1.0.0 changes"

### PR DESCRIPTION
**Pull request by perk-sumo**
This reverts commit f25e71fb3f4ce8b151f6983a25a43740415b957c.

###### Description

All the changes added to master since the revert:
https://github.com/SumoLogic/sumologic-kubernetes-collection/compare/3cdc6acd716cfa8a003dcb78948991d5dbe58d1a..revert-the-1.0-revert

This ^^ should show all the changes added in the meantime.
Please double check nothing is missing and nothing got reverted.

###### Testing performed

- [x] ci/build.sh
- [x] Redeploy fluentd and fluentd-events pods
- [x] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Comment by perk-sumo** [comment]
> Looks ok, could we strip all spaces before endline?

I thought about it and I think maybe let's not do it?
I didn't want to introduce any new changes to not make this commit even more complicated.

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -106,126 +71,279 @@ sumologic:
...
+        clientKey: ""
+        ## Path to a file containing the bearer token to use for authentication
+        bearerTokenFile: ""
+
+      ## To use additional filter plugins
+      extraFilterPluginConf: |-
+
+    ## Kubelet log configuration
+    kubelet:
+      outputConf: |-
```
Comment (vsinghal13) [inline deploy/helm/sumologic/values.yaml]: @samjsong do you want to add the `enabled` flags for the log types in this PR itself?

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -106,126 +71,279 @@ sumologic:
...
+      ## A regular expression for priority.
+      ## Matching priority will be excluded from Sumo. The logs will still be sent to FluentD.
+      excludePriorityRegex: ""
+      ## A regular expression for unit.
+      ## Matching unit will be excluded from Sumo. The logs will still be sent to FluentD.
+      excludeUnitRegex: ""
+
+    ## Systemd log configuration
+    systemd:
+      outputConf: |-
```
Comment (vsinghal13) [inline deploy/helm/sumologic/values.yaml]: same `enabled` flag

File: deploy/helm/sumologic/values.yaml [hunk 3]
```diff
@@ -106,126 +71,279 @@ sumologic:
...
+    ## the API version.
+    # watchResourceEventsOverrides:
+    #   pods: "v1"
+    #   events: "events.k8s.io/v1beta1"
+
 ## Configure metrics-server
 ## ref: https://github.com/helm/charts/blob/master/stable/metrics-server/values.yaml
 metrics-server:
-  ## Set the enabled flag to true for enabling metrics-server.
+  ## Set the enabled flag to true for enabling metrics-server.
```
Comment (vsinghal13) [inline deploy/helm/sumologic/values.yaml]: `nit`: extra space

File: deploy/helm/sumologic/values.yaml [hunk 4]
```diff
@@ -106,126 +71,279 @@ sumologic:
...
-  ## Set the enabled flag to true for enabling metrics-server.
+  ## Set the enabled flag to true for enabling metrics-server. 
   ## This is required before enabling fluentd autoscaling unless you have an existing metrics-server in the cluster.
   enabled: false
   args:
     - --kubelet-insecure-tls
     - --kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname
 
-## Configure fluent-bit
+## Configure fluent-bit
```
Comment (vsinghal13) [inline deploy/helm/sumologic/values.yaml]: `nit`: extra space

File: deploy/helm/sumologic/values.yaml [hunk 5]
```diff
@@ -280,7 +398,7 @@ fluent-bit:
     regex:
       - name: multi_line
         regex: (?<log>^{"log":"\d{4}-\d{1,2}-\d{1,2}.\d{2}:\d{2}:\d{2}.*)
-
+
```
Comment (vsinghal13) [inline deploy/helm/sumologic/values.yaml]: `nit`: extra spaces

File: deploy/helm/sumologic/templates/hpa.yaml [hunk 6]
```diff
@@ -1,17 +1,17 @@
-{{- if .Values.sumologic.fluentd.autoscaling.enabled}}
+{{- if .Values.fluentd.autoscaling.enabled}}
 apiVersion: autoscaling/v1
 kind: HorizontalPodAutoscaler
 metadata:
   name: {{ template "sumologic.fullname" . }}
   labels:
     app: {{ template "sumologic.labels.app" . }}
-    {{- include "sumologic.labels.common" . | nindent 4 }}
+    {{- include "sumologic.labels.common" . | nindent 4 }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/hpa.yaml]: trailing spaces here

Comment (perk-sumo) [reply deploy/helm/sumologic/templates/hpa.yaml]: Done :)

**Comment by sumo-drosiek** [comment]
@perk-sumo It would be easier to review if you not use force-push, please
It's hard to catch what you changed, and you can squash and force-push after all :)

**Comment by perk-sumo** [comment]
> @perk-sumo It would be easier to review if you not use force-push, please
> It's hard to catch what you changed, and you can squash and force-push after all :)

If you use the https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/509/files and check the files you've already seen as `Viewed` then GH shows only the ones that changed.
It might help you track the changes and I wont be blocked with rebasing :)

**Comment by sumo-drosiek** [comment]
> > @perk-sumo It would be easier to review if you not use force-push, please
> > It's hard to catch what you changed, and you can squash and force-push after all :)
> 
> If you use the https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/509/files and check the files you've already seen as `Viewed` then GH shows only the ones that changed.
> It might help you track the changes and I wont be blocked with rebasing :)

Rebases. It explains everything 👍

File: deploy/helm/sumologic/values.yaml (hunk 1 above)
Comment (samjsong) [reply deploy/helm/sumologic/values.yaml]: let's include these in a separate PR if we choose to do this 👍

**Pull request merged**